gitrepo/gitrepo.go

```diff
@@ -38,13 +38,63 @@ func RepoLocatedAt(path string) GitRepo {
 	return GitRepo{absoluteRoot}
 }
 
-//GetDiffForStagedFiles gets all the staged files and collects the diff section in each file
+// GetDiffForStagedFiles gets all the staged files and collects the diff section in each file
 func (repo GitRepo) GetDiffForStagedFiles() []Addition {
-	files := repo.stagedFiles()
-	result := make([]Addition, len(files))
-	for i, file := range files {
-		data := repo.fetchStagedDiff(file)
-		result[i] = NewAddition(file, data)
+	stagedContent := repo.executeRepoCommand("git", "diff", "--staged")
+	content := strings.TrimSpace(string(stagedContent))
+	lines := strings.Split(content, "\n")
+	result := make([]Addition, 0)
+
+	if len(lines) < 1 {
+		return result
+	}
+
+	// Standard git diff header pattern
+	// ref: https://git-scm.com/docs/diff-format#_generating_patches_with_p
+	headerRegex := regexp.MustCompile("^diff --git a/([^ ]+) ")
```

Comment:
`diff --git a/([^ ]+)` assumes that the file in the diff is an existing one. I think you should match `b/([^ ]+)` for and use it when a = /dev/null

[^ ]+ is an interesting pattern. While this should address all *sane* cases, a file could have spaces in their name, I don't know how to address/fix for that case, just highlighting it here for the sake of record.

Reply:
Looks like i was mistaken. `/dev/null` does not show up in the diff line in the current version of git I'm using.